code/modules/vehicles/forklift/modules/departmental_machinery.dm

```diff
@@ -0,0 +1,99 @@
+/datum/forklift_module/department_machinery
+	name = "Departmental Machinery"
+	current_selected_typepath = /obj/structure/frame/machine
+	available_builds = list(
+		/obj/structure/frame/machine,
+		/obj/structure/frame/computer,
+	) // Populated on New
+	resource_price = list(
+		/obj/structure/frame/machine = list(
+			/datum/material/iron = SHEET_MATERIAL_AMOUNT * 5,
+		),
+		/obj/structure/frame/computer = list(
+			/datum/material/iron = SHEET_MATERIAL_AMOUNT * 5,
+		),
+	) // Populated on New
+	build_length = 5 SECONDS
+	turf_place_on_top = TRUE
+	show_name_on_change = TRUE
+	deconstruction_cooldown = 5 SECONDS
+	var/bitflag_to_use = DEPARTMENT_BITFLAG_ENGINEERING
+	var/list/machinery_price = list(
+		/datum/material/iron = (SHEET_MATERIAL_AMOUNT * 5) + 50, // 5x cable coils + 5 sheets of metal
+		/datum/material/glass = 25, // 5x cable coils
+	)
+	var/list/computer_price = list(
+		/datum/material/iron = (SHEET_MATERIAL_AMOUNT * 5) + 50, // 5x cable coils + 5 sheets of metal
+		/datum/material/glass = 25, // 5x cable coils
+	)
+
+/datum/forklift_module/department_machinery/New()
+	. = ..()
+	for(var/typepath in subtypesof(/datum/design/board))
+		var/datum/design/board/checked_path = typepath
+		var/obj/item/circuitboard/board_checked = initial(checked_path.build_path)
+		if(initial(checked_path.departmental_flags) & bitflag_to_use)
+			available_builds += initial(board_checked.build_path)
+			if(ispath(initial(board_checked.build_path), /obj/machinery/computer))
+				resource_price[initial(board_checked.build_path)] = computer_price
+			else
+				resource_price[initial(board_checked.build_path)] = machinery_price
+		continue
+
+/datum/forklift_module/department_machinery/on_scrollwheel(mob/source, atom/A, scrolled_up)
+	var/list/available_paths_real = list(
+		/obj/structure/frame/machine,
+		/obj/structure/frame/computer,
+	)
+	var/datum/techweb/station_techweb = SSresearch.techwebs_assoc_lookup[TECHWEB_STATION]
```

Comment:
This is not good, the techweb should be stored locally and referred to there. This is hardcoding the machine to always use the station techweb, you never know if someone in the future would want to give one to Oldstation, Golems, Cybersun Ship, or whatever else exists in the future.

Reply:
oranges's design document does not specify support for oldstation/out of station locations, someone can patch that support in in the future

Reply:
Oranges design document doesn't specify techwebs at all but it is bad to hardcode techwebs in general, I've been pushing to dehardcode them for a while and this is gonna regress it. Please change it.